gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java

```diff
@@ -184,6 +188,79 @@ public void createBucket(String bucketName, CreateBucketOptions options) throws
     }
   }
 
+  /** See {@link GoogleCloudStorage#listBucketNames()} for details about expected behavior. */
+  @Override
+  public List<String> listBucketNames() throws IOException {
+    logger.atFiner().log("listBucketNames()");
+    List<Bucket> allBuckets = listBucketsInternal();
+    List<String> bucketNames = new ArrayList<>(allBuckets.size());
+    for (Bucket bucket : allBuckets) {
+      bucketNames.add(bucket.getName());
+    }
+    return bucketNames;
+  }
+
+  /** See {@link GoogleCloudStorage#listBucketInfo()} for details about expected behavior. */
+  @Override
+  public List<GoogleCloudStorageItemInfo> listBucketInfo() throws IOException {
+    logger.atFiner().log("listBucketInfo()");
+    List<Bucket> allBuckets = listBucketsInternal();
+    List<GoogleCloudStorageItemInfo> bucketInfos = new ArrayList<>(allBuckets.size());
+    for (Bucket bucket : allBuckets) {
+      bucketInfos.add(createItemInfoForBucket(new StorageResourceId(bucket.getName()), bucket));
+    }
+    return bucketInfos;
+  }
+
+  /**
+   * Shared helper for actually dispatching buckets list API calls and accumulating paginated
+   * results; these can then be used to either extract just their names, or to parse into full
+   * GoogleCloudStorageItemInfos.
+   */
+  private List<Bucket> listBucketsInternal() throws IOException {
+    logger.atFiner().log("listBucketsInternal()");
+    checkNotNull(storageOptions.getProjectId(), "projectId must not be null");
+    List<Bucket> allBuckets = new ArrayList<>();
+    try {
+      Page<Bucket> buckets =
+          storage.list(
+              BucketListOption.pageSize(storageOptions.getMaxListItemsPerCall()),
```

Comment:
What is the max allowed?

Reply:
1000 for bucket fields without acl and 200 with acl. We use non-acl fields so it is 1000 for this use case.